lib/resolver.js

```diff
@@ -1,39 +1,49 @@
-
 const Joi = require('@hapi/joi');
 const Util = require('util');
 const Hoek = require('@hapi/hoek');
+const Bourne = require('@hapi/bourne');
 
 class SchemaResolver {
-    constructor(root, { subSchemas, types, refineType, strictMode, extensions = [] }) {
+    constructor(root, { subSchemas, refineType, strictMode, extensions = [] }) {
         this.root = root;
-        this.types = types;
         this.subSchemas = subSchemas;
         this.refineType = refineType;
         this.strictMode = strictMode;
 
-        extensions.push({
-            name: 'allOf',
-            rules: [{
-                name: 'items',
-                params: {
-                    items: Joi.array().items(Joi.object()).required()
-                },
-                validate(params, value, state, options) {
-                    options = Hoek.applyToDefaults(options, { allowUnknown: true });
-
-                    for (const schema of params.items) {
-                        const { error } = schema.validate(value, options);
-                        if (error) {
-                            const details = error.details[0];
-                            return this.createOverrideError(details.type, details.context, { path : details.path }, options, details.message);
+        this.joi = Joi.extend(
+            {
+                type: 'object',
+                base: Joi.object(),
+                coerce: {
+                    from: 'string',
+                    method(value) {
+                        if (typeof value !== 'string' || (value[0] !== '{' && !/^\s*\{/.test(value))) {
+                            return;
                         }
+
+                        try {
+                            return { value: Bourne.parse(value) };
+                        } catch (ignoreErr) { }
```

Comment:
What is the impact of ignoring this error?

Reply:
Same as if the input is not string or does not validate with the regular expression. The coerce method returns `undefined`, which becomes the object or array value. This part of the code is actually adopted from Joi version 16 release notes: https://github.com/hapijs/joi/issues/2037